File: WdtcCore/src/main/java/org/wdt/wdtc/auth/yggdrasil/AuthlibInjector.java
package org.wdt.wdtc.auth.yggdrasil;

import org.wdt.utils.io.FileUtils;
import org.wdt.wdtc.download.DownloadTask;
import org.wdt.wdtc.manger.FileManger;
import org.wdt.wdtc.utils.UrlUtils;
import org.wdt.wdtc.utils.gson.JSONObject;

import java.io.IOException;
import java.util.jar.JarInputStream;

public class AuthlibInjector {
    private static final String BMCL_AUTHLIB_INJECTOR_LATEST_JSON = "https://bmclapi2.bangbang93.com/mirrors/authlib-injector/artifact/latest.json";
    private static final String AUTHLIB_INJECTOR = "https://authlib-injector.yushi.moe/";

    public static void downloadauthlibInjector() throws IOException {
        DownloadTask.StartDownloadTask(getBmclAuthlibInjectorLatestJsonObject().getString("download_url"), FileManger.getAuthlibInjector());
    }

    public static void updateAuthlibInjector() throws IOException {
        if (FileUtils.isFileExists(FileManger.getAuthlibInjector())) {
            String LatestVersionNumber = getBmclAuthlibInjectorLatestJsonObject().getString("version");
            String PresentVersionNumber = new JarInputStream(FileUtils.newInputStream(FileManger.getAuthlibInjector()))
                    .getManifest().getMainAttributes().getValue("Implementation-Version");
            if (!PresentVersionNumber.equals(LatestVersionNumber)) {
                downloadauthlibInjector();
            }
        } else {
            downloadauthlibInjector();
        }
    }

    private static JSONObject getBmclAuthlibInjectorLatestJsonObject() throws IOException {
        return JSONObject.parseObject(UrlUtils.getUrlToString(BMCL_AUTHLIB_INJECTOR_LATEST_JSON));
    }
}

File: core/core-impl/src/main/kotlin/org/wdt/wdtc/core/impl/auth/accounts/yggdrasil/AuthlibInjector.kt
@file:JvmName("AuthlibInjector")

package org.wdt.wdtc.core.impl.auth.accounts.yggdrasil

import org.wdt.utils.gson.getString
import org.wdt.utils.gson.parseJsonObject
import org.wdt.utils.io.toStrings
import org.wdt.wdtc.core.openapi.manger.authlibInjector
import org.wdt.wdtc.core.openapi.utils.compareFile
import org.wdt.wdtc.core.openapi.utils.startDownloadTask
import org.wdt.wdtc.core.openapi.utils.toFileData
import org.wdt.wdtc.core.openapi.utils.toURL

private const val BMCL_AUTHLIB_INJECTOR_LATEST_JSON =
	"https://bmclapi2.bangbang93.com/mirrors/authlib-injector/artifact/latest.json"
private const val AUTHLIB_INJECTOR = "https://authlib-injector.yushi.moe/"

suspend fun downloadAuthlibInjector() {
	BMCL_AUTHLIB_INJECTOR_LATEST_JSON.toURL().toStrings().parseJsonObject().getString("download_url").toURL().let {
		if (authlibInjector compareFile it.toFileData()) {
				startDownloadTask(it, authlibInjector)
			}
		}
	
}

